tiledb/sm/config/config.cc

```diff
@@ -754,15 +756,28 @@ Status Config::sanity_check(
   int64_t vint64 = 0;
   int chkno = -1;
 
+  // Check for RestProfile-specific parameters and warn users accordingly.
+  // Note: users are allowed to _fetch_ these parameters, but they must
+  // first be set on an immutable RestProfile.
+
+  // What happens if the user creates the Profile but doesn't save it?
+  // Maybe add a bool saved_ flag in RP to track if the profile's been saved.
+  if (param == "rest.password" || param == "rest.payer_namespace" ||
```

Comment:
Rather than see this in line here I think we'll be better off if the `RestProfile` class has a list of its supported parameters which we can check against.

Reply:
This code block is not present anymore.